src/containers/DAO/pages/DAOPage/Proposals/ProposalDetail/VoteAttributes/VoteAttributes.tsx

```diff
@@ -0,0 +1,169 @@
+import { capitalizeString } from 'lib/utils/string';
+import { secondsToDhms, formatDateTime } from 'lib/utils/datetime';
+import { formatTotalAmountOfTokenMetadata } from '../../PropsalsTable/ProposalsTable.helpers';
+import { truncateWalletAddress } from 'lib/utils';
+
+// - Types
+import { VoteAttribute } from './VoteAttributes.types';
+
+// Constants
+import { CURRENCY } from 'constants/currency';
+import { VOTE_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT } from './VoteAttributes.constants';
+
+//- Style Imports
+import styles from './VoteAttributes.module.scss';
+
+type VoteAttributesProps = {
+	proposal?: Proposal;
+	metadata?: TokenMetaData;
+};
+
+export const VoteAttributes: React.FC<VoteAttributesProps> = ({
+	proposal,
+	metadata,
+}) => {
+	const [isCollapsed, toggleCollapsed] = useState<boolean>(true);
+
+	const initialVisibleAttributesCount: number = useMemo(() => {
+		const isTablet = window.innerWidth > 414 && window.innerWidth < 768;
+		const isMobile = window.innerWidth <= 414;
+
+		if (isMobile) return VOTE_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.MOBILE;
+		if (isTablet) return VOTE_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.TABLET;
+		return VOTE_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.DESKTOP;
+	}, []);
+
+	const attributes: VoteAttribute[] = useMemo(() => {
+		const result: VoteAttribute[] = [];
+
+		if (!proposal || !metadata) {
+			return result;
+		}
+
+		const wild = formatTotalAmountOfTokenMetadata(metadata);
+
+		// TODO: Should align the attributes
+		result.push({
+			label: 'Status',
+			value: capitalizeString(proposal.state),
+		});
+
+		result.push({
+			label: 'Time Remaining',
+			value: secondsToDhms(moment(proposal.end).diff(moment()) / 1000) || '-',
+		});
+
+		result.push({
+			label: 'Type',
+			value: capitalizeString(proposal.type),
+		});
+
+		result.push({
+			label: 'Amount',
+			value: wild ? wild + ' ' + CURRENCY.WILD : '-',
+		});
+
+		result.push({
+			label: 'Voting Started',
+			value: formatDateTime(proposal.start, 'M/D/YYYY h:m A Z') || '-',
+		});
+
+		result.push({
+			label: 'Voting Ends',
+			value: formatDateTime(proposal.start, 'M/D/YYYY h:m A Z') || '-',
+		});
+
+		result.push({
+			label: 'Voting System',
+			value: '-',
+		});
+
+		result.push({
+			label: 'Execution Criteria',
+			value: '-',
+		});
+
+		result.push({
+			label: 'Creator',
+			value: truncateWalletAddress(proposal.author, 4) || '-',
+		});
+
+		result.push({
+			label: 'Source of Funds',
+			value: '-',
+		});
+
+		result.push({
+			label: 'Recipient',
+			value: truncateWalletAddress(metadata.recipient, 4) || '-',
+		});
+
+		result.push({
+			label: 'Votes Submitted',
+			value: '-',
+		});
+
+		return result;
+	}, [proposal, metadata]);
+
+	const initialHiddenAttributesCount: number = useMemo(() => {
+		if (attributes.length > initialVisibleAttributesCount) {
+			return attributes.length - initialVisibleAttributesCount;
+		}
+
+		return 0;
+	}, [attributes, initialVisibleAttributesCount]);
```

Comment:
I think `useMemo` might be less efficient here, compared to calculating the value per render. `useMemo` has some background overhead from having to cache a value and recall it, so in situations where the calculation isn't resource intensive it doesn't provide a performance boost (sometimes the opposite).

Could instead be
```js
return Math.max(attributes.length - initialVisibleAttributesCount, 0)
```